src/ZLogger.MessagePack/MessagePackZLoggerFormatter.cs

```diff
@@ -0,0 +1,155 @@
+        // "Exception"
+        static readonly byte[] ExceptionKey = { 0b10100000 | 9, 69, 120, 99, 101, 112, 116, 105, 111, 110 };
+
+        // "Name"
+        static readonly byte[] NameKey = { 0b10100000 | 4, 78, 97, 109, 101 };
+        // "Message"
+        static readonly byte[] MessageKey = { 0b10100000 | 7, 77, 101, 115, 115, 97, 103, 101 };
+        // "StackTrace"
+        static readonly byte[] StackTraceKey = { 0b10100000 | 10, 83, 116, 97, 99, 107, 84, 114, 97, 99, 101 };
+        // "InnerException"
+        static readonly byte[] InnerExceptionKey = { 0b10100000 | 14, 73, 110, 110, 101, 114, 69, 120, 99, 101, 112, 116, 105, 111, 110 };
+
+        // "Trace"
+        static readonly byte[] Trace = { 0b10100000 | 5, 84, 114, 97, 99, 101 };
+        // "Debug"
+        static readonly byte[] Debug = { 0b10100000 | 5, 68, 101, 98, 117, 103 };
+        // "Information"
+        static readonly byte[] Information = { 0b10100000 | 11, 73, 110, 102, 111, 114, 109, 97, 116, 105, 111, 110 };
+        // "Warning"
+        static readonly byte[] Warning = { 0b10100000 | 7, 87, 97, 114, 110, 105, 110, 103 };
+        // "Error"
+        static readonly byte[] Error = { 0b10100000 | 5, 69, 114, 114, 111, 114 };
+        // "Critical"
+        static readonly byte[] Critical = { 0b10100000 | 8, 67, 114, 105, 116, 105, 99, 97, 108 };
+        // "None"
+        static readonly byte[] None = { 0b10100000 | 4, 78, 111, 110, 101 };
+
+        public MessagePackSerializerOptions MessagePackSerializerOptions { get; set; } = MessagePackSerializerOptions.Standard;
+        public string MessagePropertyName { get; set; } = "Message";
+        public string PayloadPropertyName { get; set; } = "Payload";
+
+        public void FormatLogEntry<TEntry, TPayload>(
+            IBufferWriter<byte> writer,
+            TEntry entry,
+            TPayload payload,
+            ReadOnlySpan<byte> utf8Message)
+            where TEntry : IZLoggerEntry
+        {
+            var messagePackWriter = new MessagePackWriter(writer);
+            var hasPayload = payload != null;
+            var hasException = entry.LogInfo.Exception != null;
+
+            messagePackWriter.WriteMapHeader(6 + (hasException ? 1 : 0) + (hasPayload ? 1 : 0));
+
+            messagePackWriter.WriteRaw(CategoryNameKey);
+            messagePackWriter.Write(entry.LogInfo.CategoryName);
+
+            messagePackWriter.WriteRaw(LogLevelKey);
+            messagePackWriter.WriteRaw(EncodedLogLevel(entry.LogInfo.LogLevel));
+
+            messagePackWriter.WriteRaw(EventIdKey);
+            messagePackWriter.WriteInt32(entry.LogInfo.EventId.Id);
+
+            messagePackWriter.WriteRaw(EventIdNameKey);
+            messagePackWriter.Write(entry.LogInfo.EventId.Name);
+
+            messagePackWriter.WriteRaw(TimestampKey);
+            messagePackWriter.Write(entry.LogInfo.Timestamp.UtcDateTime);
```

Comment:
The timestamp is in MessagePack timestamp extension format. Is this correct?
If we often convert to JSON, is this a little inconvenient(?)